Src/FluentAssertions/Equivalency/EnumEqualityStep.cs

```diff
@@ -56,5 +52,65 @@ public bool CanHandle(IEquivalencyValidationContext context, IEquivalencyAsserti
 
             return true;
         }
+
+        private static void HandleByValue(IEquivalencyValidationContext context)
+        {
+            decimal? subjectsUnderlyingValue = ExtractDecimal(context.Subject);
+            decimal? expectationsUnderlyingValue = ExtractDecimal(context.Expectation);
+
+            Execute.Assertion
+                .ForCondition(subjectsUnderlyingValue == expectationsUnderlyingValue)
+                .FailWith(() =>
+                {
+                    string subjectsName = EnumDescription(context.Subject, subjectsUnderlyingValue);
+                    string expectationName = EnumDescription(context.Expectation, expectationsUnderlyingValue);
+
+                    return new FailReason($"Expected enum to equal {expectationName} by value{{reason}}, but found {subjectsName}.");
+                });
+        }
+
+        private static void HandleByName(IEquivalencyValidationContext context)
+        {
+            string subject = context.Subject.ToString();
```

Comment:
🤔 What if the subject or expectation is `null`?

Reply:
The enums can only be `null` if they are nullable enums.
`CanHandle` checks that the types of `subject` and `expectation` has `IsEnum` which nullable enums do not.

Reply:
Actually, it was a bug. In that case NRE was thrown. @jnyrup you missed the fact that `CanHandle` checks if `subject` **or** `expectaction` is enum, which makes it possible for one side of the comparison to be null. I have added tests for both cases.